src/main/java/com/mg105/use_cases/ReplayGenerator.java

```diff
@@ -33,6 +35,23 @@ private void attributeInheritance() {
         }
     }
 
+    /**
+     * Revive all characters and add them to the list of alive characters
+     */
+    public void reviveCharacters() {
+        MegaPotion potion = new MegaPotion();
+        potion.consume(state, PartyConstants.ALL_PARTY_MEMBER_NAMES[0]);
+
+        MegaPotion potion2 = new MegaPotion();
+        potion2.consume(state, PartyConstants.ALL_PARTY_MEMBER_NAMES[1]);
+
+        MegaPotion potion3 = new MegaPotion();
+        potion3.consume(state, PartyConstants.ALL_PARTY_MEMBER_NAMES[2]);
+
+        MegaPotion potion4 = new MegaPotion();
+        potion4.consume(state, PartyConstants.ALL_PARTY_MEMBER_NAMES[3]);
+    }
+
```

Comment:
This isn't the best way to do this (consider if one character has been upgraded so much that it has an hp greater than the mega potion value).

What I would do have a constant full heal value which is extremely high and loop over the member names and increase the battle character by the constant hp (if you do not know how the get the battle character from the name look at the gamestate public methods and if you do not know how to increase a characters hp look at the battle character public method)

Alternatively you could also make a function in battle character that fully heals a character

Reply:
This looks good now but there is no point in keeping the megapoition code (unless ur using it to but the character back into the party list from the fainted list without doing it manually)